fmriprep/interfaces/patches.py

```diff
@@ -12,49 +12,110 @@
 
 from numpy.linalg.linalg import LinAlgError
 from nipype.algorithms import confounds as nac
+from nipype.interfaces.base import (File, traits)
+from nipype.interfaces.mixins import reporting
 
 
-class RobustACompCor(nac.ACompCor):
-    """
-    Runs aCompCor several times if it suddenly fails with
-    https://github.com/poldracklab/fmriprep/issues/776
+class RetryCompCorInputSpecMixin(reporting.ReportCapableInputSpec):
+    out_report = File('report.html', usedefault=True, hash_files=False,
+                      desc='filename for warning HTML snippet')
+    # 'NaN` by default
+    failure_mode = traits.Enum(
+        'NaN', 'error',
```

Comment:
Is there any reason not to make error the default?

```suggestion
        'error', 'NaN',  # 'error' by default
```

Reply:
It doesn't really matter, since I'm already specifying `NaN` in the workflow. I'll just drop this trait altogether, as it defaults to error in the `CompCorInputSpec`.